src/main/java/teammates/logic/core/AccountsLogic.java

```diff
@@ -115,8 +116,13 @@ public StudentAttributes joinCourseForStudent(String registrationKey, String goo
      * Joins the user as an instructor and sets the institute if it is not null.
      * If the given instructor is null, the instructor is given the institute of an existing instructor of the same course.
      */
-    public InstructorAttributes joinCourseForInstructor(String encryptedKey, String googleId, String institute)
+    public InstructorAttributes joinCourseForInstructor(String encryptedKey, String googleId, String institute, String mac)
             throws InvalidParametersException, EntityDoesNotExistException, EntityAlreadyExistsException {
+
+        if (institute != null && !StringHelper.isCorrectSignature(institute, mac)) {
+            throw new InvalidParametersException("Institute authentication failed.");
```

Comment:
Nothing wrong here - you throw an Invalid Parameter Exception when the hash doesn't match.